datadog/plugins/init.sls

```diff
@@ -0,0 +1,20 @@
+{% from "datadog/map.jinja" import datadog with context %}
+
+include:
+  - ..service
+
+datadog_plugin_configuration_directory:
+  file.directory:
+    - name: /etc/dd-agent/conf.d
+
+{% for name, settings in salt.pillar.get('datadog:integrations', {}).items() %}
```

Comment:
As we discussed, `overrides` for a name might be better, and merging the root key in can cause confusion and obscure the separation between forced pillar data entry and overridable defaults, so the convenience benefit probably doesn't outweigh the disadvantages.